apps/omg_child_chain/lib/omg_child_chain/gas_price/strategy/legacy_gas_strategy.ex

```diff
@@ -0,0 +1,239 @@
+    # Its result needs to be applied to the upcoming block immediately.
+    GenServer.call(__MODULE__, {:recalculate, latest})
+  end
+
+  #
+  # GenServer callbacks
+  #
+
+  @doc false
+  @impl GenServer
+  def handle_call(:get_price, _, state) do
+    {:reply, {:ok, state.gas_price}, state}
+  end
+
+  @doc false
+  @impl GenServer
+  def handle_call({:recalculate, latest}, _, state) do
+    state = recalculate_state(latest, state)
+
+    _ = Logger.info("#{__MODULE__}: Gas price recalculated. New price: #{inspect(state.gas_price)}")
+
+    {:reply, :ok, state}
+  end
+
+  #
+  # Internal implementations
+  #
+
+  defp recalculate_state(latest, state) do
+    cond do
+      latest.parent_height - state.last_parent_height < state.eth_gap_without_child_blocks ->
+        state
+
+      !blocks_to_mine(
+        latest.blocks,
+        latest.mined_child_block_num,
+        latest.formed_child_block_num,
+        latest.child_block_interval
+      ) ->
+        state
+
+      true ->
+        gas_price =
+          calculate_gas_price(
+            latest.mined_child_block_num,
+            latest.formed_child_block_num,
+            state.last_mined_child_block_num,
+            state.gas_price,
+            state.gas_price_raising_factor,
+            state.gas_price_lowering_factor,
+            state.max_gas_price
+          )
+
+        state = Map.update!(state, :gas_price, gas_price)
+        _ = Logger.debug("using new gas price '#{inspect(state.gas_price)}'")
+
+        case state.last_mined_child_block_num < latest.mined_child_block_num do
+          true ->
+            state
+            |> Map.update!(:last_parent_height, latest.parent_height)
+            |> Map.update!(:last_mined_child_block_num, latest.mined_child_block_num)
+
+          false ->
+            state
+        end
+    end
+  end
+
+  # Calculates the gas price basing on simple strategy to raise the gas price by gas_price_raising_factor
+  # when gap of mined parent blocks is growing and droping the price by gas_price_lowering_factor otherwise
+  @spec calculate_gas_price(
+          non_neg_integer(),
+          non_neg_integer(),
+          non_neg_integer(),
+          pos_integer(),
+          float(),
+          float(),
+          pos_integer()
+        ) :: pos_integer()
+  defp calculate_gas_price(
+         mined_child_block_num,
+         formed_child_block_num,
+         last_mined_child_block_num,
+         gas_price,
+         raising_factor,
+         lowering_factor,
+         max_gas_price
+       ) do
+    blocks_needs_be_mined? = blocks_needs_be_mined?(formed_child_block_num, mined_child_block_num)
+    new_blocks_mined? = new_blocks_mined?(mined_child_block_num, last_mined_child_block_num)
+
+    multiplier =
+      case {blocks_needs_be_mined?, new_blocks_mined?} do
+        {false, _} -> 1.0
+        {true, false} -> raising_factor
+        {_, true} -> lowering_factor
```

Comment:
minor: I think it is easier to read with `{true, true} -> lowering_factor` instead of `_`.

Reply:
Updated https://github.com/omgnetwork/elixir-omg/pull/1575/commits/0401a8faf69d58a3606390304d12ff1c67f7ee9f thanks!